bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -268,6 +279,15 @@ protected void addProvider(Provider<E> provider) {
         }
     }
 
+    /**
+     * Returns the key for a given element
+     *
+     * @param element
+     *            element
+     * @return key (must not be null)
+     */
+    protected abstract K getKey(E element);
```

Comment:
Hm, it would be great if we could use `getUID()` for the key, because we would like a unique ID.
For the item itself it would be easy, the name will be the unique. All the other classes already provide a `getUID()` method.
BUT: The "unique" identifier for the link is provided by `getID()` and the stuff that is linked to an item will be returned by `getUID()`. This is a little bit confusing IMHO.